File: data/readable_format/cpp/file1448.cpp
#include <array>
#include <gtest/gtest.h>

// Define the sizes namespace directly in this file, without using a separate header file.
namespace sizes {
    constexpr int Ndim = 3;
    constexpr std::array<int, Ndim> mesh = {100, 100, 100};
    constexpr int meshx = mesh[0];
    constexpr int meshy = mesh[1];
    constexpr int meshz = mesh[2];
}

// Test for Ndim
TEST(SizesTest, TestNdim) {
    EXPECT_EQ(sizes::Ndim, 3);
}

// Test for mesh array values
TEST(SizesTest, TestMeshArray) {
    EXPECT_EQ(sizes::mesh[0], 100);
    EXPECT_EQ(sizes::mesh[1], 100);
    EXPECT_EQ(sizes::mesh[2], 100);
}

// Test for meshx, meshy, meshz
TEST(SizesTest, TestMeshXYZ) {
    EXPECT_EQ(sizes::meshx, 100);
    EXPECT_EQ(sizes::meshy, 100);
    EXPECT_EQ(sizes::meshz, 100);
}

// The main function that runs all tests
int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/readable_format/fortran/file1448.f90
module sizes
  implicit none
  private

  integer, parameter, public :: Ndim=3
  integer, dimension(Ndim), parameter, public :: mesh=(/ 100, 100, 100 /)
  integer, parameter, public :: meshx=mesh(1)
  integer, parameter, public :: meshy=mesh(2)
  integer, parameter, public :: meshz=mesh(3)
end module sizes

program test_sizes
  use sizes
  implicit none

  logical :: test_passed

  ! Test Ndim
  test_passed = (Ndim == 3)
  if (test_passed) then
      print *, "Ndim test passed"
  else
      print *, "Ndim test failed"
  end if

  ! Test mesh array and individual meshx, meshy, meshz
  test_passed = all(mesh == [100, 100, 100]) .and. &
                meshx == 100 .and. meshy == 100 .and. meshz == 100
  if (test_passed) then
      print *, "Mesh size tests passed"
  else
      print *, "Mesh size tests failed"
  end if
end program test_sizes